### ISSUE TITLE
twisted.conch.test.test_filetransfer.TestOurServerOurClient.testOpenFileAttributes (and perhaps others) leaks file descriptors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1789]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1789|
|Type|defect|
|Created|2006-06-04 15:48:50Z|

Running ```trial -u twisted.conch.test.test_filetransfer.TestOurServerOurClient.testOpenFileAttributes``` will eventually fail with an error about there being too many open files.

<details><summary>Searchable metadata</summary>

```
trac-id__1789 1789
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1149436130000000 1149436130000000
changetime__1211623477000000 1211623477000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

twisted.conch.test.test_cftp exhibits a similar problem.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

It will be solved by [#3213](https://github.com/twisted/twisted/issues/3213).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [23695]) Merge filetransfer-leak-3213

Author: therve
Reviewer: exarkun
Fixes [#3213](https://github.com/twisted/twisted/issues/3213)
Fixes [#1789](https://github.com/twisted/twisted/issues/1789)

Implement connectionLost on twisted.conch.filetransfer.FileTransferServer to
close all opened files and directories: previously, if a connection died
unexpectedly, all opened files were leaked.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
